timeseries/src/autogluon/timeseries/models/gluonts/torch/models.py

```diff
@@ -0,0 +1,167 @@
+        """Get GluonTS specific constructor arguments for estimator objects, an alias to
+        `self._get_model_params` for better readability."""
+        init_kwargs = self._get_model_params()
+
+        # GluonTS does not handle context_length=1 well, and sets the context to only prediction_length
+        # we set it to a minimum of 10 here.
+        init_kwargs["context_length"] = max(10, init_kwargs.get("context_length", self.prediction_length))
+
+        return init_kwargs
+
+    def _get_estimator(self) -> GluonTSPyTorchLightningEstimator:
+        """Return the GluonTS Estimator object for the model"""
+
+        # As GluonTSPyTorchLightningEstimator objects do not implement `from_hyperparameters` convenience
+        # constructors, we re-implement the logic here.
+        # we translate the "epochs" parameter to "max_epochs" for consistency in the AbstractGluonTSModel
+        # interface
+
+        init_args = self._get_estimator_init_args()
+
+        trainer_kwargs = {}
+        epochs = init_args.get("max_epochs", init_args.get("epochs"))
+        callbacks = init_args.get("callbacks", [])
+
+        if epochs is not None:
+            trainer_kwargs.update({"max_epochs": epochs})
+        trainer_kwargs.update({"callbacks": callbacks, "enable_progress_bar": False})
+
+        return from_hyperparameters(
+            self.gluonts_estimator_class,
+            trainer_kwargs=trainer_kwargs,
+            **init_args,
+        )
+
+    def _get_callbacks(self, time_limit: int, *args, **kwargs) -> List[Callable]:
+        return [Timer(timedelta(seconds=time_limit))] if time_limit is not None else []
+
+    def _fit(
+        self,
+        train_data: TimeSeriesDataFrame,
+        val_data: Optional[TimeSeriesDataFrame] = None,
+        time_limit: int = None,
+        **kwargs,
+    ) -> None:
+        verbosity = kwargs.get("verbosity", 2)
+        for pl_logger in pl_loggers:
+            pl_logger.setLevel(logging.ERROR if verbosity <= 3 else logging.INFO)
+        super()._fit(train_data=train_data, val_data=val_data, time_limit=time_limit, **kwargs)
+
+    def save(self, path: str = None, **kwargs) -> str:
+        # we flush callbacks instance variable if it has been set. it can keep weak references
+        # which breaks training
+        self.callbacks = []
+        return super().save(path, **kwargs)
+
+    @classmethod
+    def load(cls, path: str, reset_paths: bool = True, verbose: bool = True) -> "AbstractGluonTSModel":
+        with torch_warning_filter():
+            model = load_pkl.load(path=path + cls.model_file_name, verbose=verbose)
+            if reset_paths:
+                model.set_contexts(path)
+            model.gts_predictor = GluonTSPyTorchPredictor.deserialize(Path(path) / cls.gluonts_model_path)
+        return model
+
+
+class DeepARModel(AbstractGluonTSPyTorchModel):
+    gluonts_estimator_class: Type[GluonTSPyTorchLightningEstimator] = DeepAREstimator
```

Comment:
Could add these TODOs here?
- TODO: Add docstring for the hyperparameters
- TODO: Enable static features